### PR TITLE
ENH add build 2 host migrator

### LINKED ISSUE #1268: Run a mini-migrator to rename build->host before osx-arm64 migrators
For eg: https://github.com/regro-cf-autotick-bot/jupyter_core-feedstock/blob/0088e71f7e8a7b7998bef034d3059239a1b1926a/recipe/meta.yaml#L20

### PR DESCRIPTION
**Pull request by beckermr**
This PR closes #1268.

cc @isuruf

### CODE REVIEW

File: conda_forge_tick/migrators/cross_compile.py [hunk 1]
```diff
@@ -212,3 +209,46 @@ def migrate(self, recipe_dir: str, attrs: "AttrsTypedDict", **kwargs: Any) -> No
...
+class Build2HostMigrator(MiniMigrator):
+    post_migration = False
+
+    def filter(self, attrs: "AttrsTypedDict", not_bad_str_start: str = "") -> bool:
+        build_reqs = attrs.get("requirements", {}).get("build", set())
+        host_reqs = attrs.get("requirements", {}).get("host", set())
+        if (
+            len(attrs.get("output_names", [])) <= 1
+            and build_reqs
+            and not host_reqs
```
Comment (isuruf) [inline conda_forge_tick/migrators/cross_compile.py:223]:
```suggestion
            and 'host' not in attrs.get("requirements", {})
```

Comment (beckermr) [reply conda_forge_tick/migrators/cross_compile.py:223]: This change is wrong. The top-level requirements in the node attrs always has `host` even if host is empty. This is not the recipe requirements.

See here: 

https://github.com/regro/cf-graph-countyfair/blob/ddde68f0f61f1a8f2e54b865bb477716d460e125/node_attrs/jupyter_core.json#L1471

for example.

File: conda_forge_tick/migrators/cross_compile.py [hunk 2]
```diff
@@ -212,3 +209,46 @@ def migrate(self, recipe_dir: str, attrs: "AttrsTypedDict", **kwargs: Any) -> No
...
+
+
+class Build2HostMigrator(MiniMigrator):
+    post_migration = False
+
+    def filter(self, attrs: "AttrsTypedDict", not_bad_str_start: str = "") -> bool:
+        build_reqs = attrs.get("requirements", {}).get("build", set())
+        host_reqs = attrs.get("requirements", {}).get("host", set())
+        if (
+            len(attrs.get("output_names", [])) <= 1
```
Comment (beckermr) [inline conda_forge_tick/migrators/cross_compile.py:221]:
```suggestion
            len(attrs.get("outputs_names", [])) <= 1
```

File: conda_forge_tick/migrators/cross_compile.py (hunk 1 above)
Comment (beckermr) [reply conda_forge_tick/migrators/cross_compile.py:223]: woops resolved that too soon! sorry!

Comment (beckermr) [reply conda_forge_tick/migrators/cross_compile.py:223]: I pushed a change just now that does what I think you were going for. It makes sure that the recipe doesn't have any host section as opposed to an empty one.

Comment (isuruf) [reply conda_forge_tick/migrators/cross_compile.py:223]: Thanks

File: conda_forge_tick/migrators/cross_compile.py [hunk 3]
```diff
@@ -212,3 +209,47 @@ def migrate(self, recipe_dir: str, attrs: "AttrsTypedDict", **kwargs: Any) -> No
...
+
+class Build2HostMigrator(MiniMigrator):
+    post_migration = False
+
+    def filter(self, attrs: "AttrsTypedDict", not_bad_str_start: str = "") -> bool:
+        build_reqs = attrs.get("requirements", {}).get("build", set())
+        host_reqs = attrs.get("requirements", {}).get("host", set())
+        if (
+            len(attrs.get("outputs_names", [])) <= 1
+            and build_reqs
```
Comment (isuruf) [inline conda_forge_tick/migrators/cross_compile.py:222]:
```suggestion
            and "python" in build_reqs and "python" in run_reqs
```

File: conda_forge_tick/migrators/cross_compile.py [hunk 4]
```diff
@@ -212,3 +209,47 @@ def migrate(self, recipe_dir: str, attrs: "AttrsTypedDict", **kwargs: Any) -> No
...
             with open("build.sh", "w") as f:
                 f.write("".join(lines))
+
+
+class Build2HostMigrator(MiniMigrator):
+    post_migration = False
+
+    def filter(self, attrs: "AttrsTypedDict", not_bad_str_start: str = "") -> bool:
+        build_reqs = attrs.get("requirements", {}).get("build", set())
+        host_reqs = attrs.get("requirements", {}).get("host", set())
```
Comment (isuruf) [inline conda_forge_tick/migrators/cross_compile.py:219]:
```suggestion
        host_reqs = attrs.get("requirements", {}).get("host", set())
        run_reqs = attrs.get("requirements", {}).get("run", set())
```

File: conda_forge_tick/migrators/cross_compile.py (hunk 1 above)
Comment (beckermr) [reply conda_forge_tick/migrators/cross_compile.py:224]: resolving now!

**Comment by beckermr** [comment]
OK @isuruf Any more comments or shall we merge?

**Review by isuruf — APPROVED** [review]

File: conda_forge_tick/migrators/__init__.py [hunk 5]
```diff
@@ -20,4 +20,5 @@
     UpdateCMakeArgsMigrator,
     GuardTestingMigrator,
     CrossPythonMigrator,
+    Build2HostMigrator,
```
Comment (isuruf) [inline conda_forge_tick/migrators/__init__.py:23]: This should be moved above `CrossPythonMigrator`

Comment (isuruf) [reply conda_forge_tick/migrators/__init__.py:23]: nvm

**Comment by CJ-Wright** [comment]
Should this go on all of our migrations? It seems like an error left over from the cb3 migration

**Comment by beckermr** [comment]
Yup. We can do that.

**Pull request merged**